include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -818,6 +819,102 @@ struct OrtApi {
                   _Inout_ OrtSessionOptions* options, _In_ const char* dim_name,
                   _In_ int64_t dim_value);
 
+  /**
+   * \param sess valid OrtSession instance
+   * \para mem_info - valid OrtMemoryInfo instance
+   * \param - out a ptr to a new instance of OrtAllocator according to the spec within mem_info
+   *         if successful
+   * \return OrtStatus or nullptr if successful
+   */
+  ORT_API2_STATUS(CreateAllocator, _In_ const OrtSession* sess, _In_ const OrtMemoryInfo* mem_info,
```

Comment:
Just remembered one thing: how does a user know what 'name' to use inside OrtMemoryInfo? In your test for cuda, you're using "Cuda" because this is the string we use [internally](https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/framework/allocator.h#L145). If any other string is used, this API will return a nullptr. The CreateCpuMemoryInfo API takes care of this by using the internal name for CPU.

Reply:
The user knows the name bc he specifies it when creating OrtMemoryInfo. Those string constants are exposed in C# and C but may need to be better documented in C.